packages/aws-rfdk/lib/core/scripts/powershell/configureCloudWatchAgent.ps1

```diff
@@ -16,62 +16,104 @@ param (
+    Write-Output "Failed to download CloudWatch agent installer."
     Exit 1
   }
-}
 
-# The below verification is tied to the version of the msi installer
-# If the installer version changes, the below verification could break and should be updated accordingly.
-# See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/verify-CloudWatch-Agent-Package-Signature.html
-if (-Not $s) {
-  $sig = Get-AuthenticodeSignature $file
-  $status = $sig | Select-Object -ExpandProperty 'Status'
-  if ( $status -ne 'Valid' ) {
-    Write-Output "CloudWatchAgent installer does not have a valid signature."
+  $cwa_installer_sig = "$env:temp\amazon-cloudwatch-agent.msi.sig"
+  try {
+    Read-S3Object -BucketName amazoncloudwatch-agent -Key windows/amd64/latest/amazon-cloudwatch-agent.msi.sig -File $cwa_installer_sig -Region us-east-1
+  } catch {
+    Write-Output "Failed to download CloudWatch agent installer signature file."
     Exit 1
   }
 
-  # CA Certificate is:
-  #  - Subject: CN=DigiCert Timestamp Responder, O=DigiCert, C=US
-  #  - Valid: [Not After] 10/21/2024 7:00:00 PM
-  #  - Thumbprint: 614D271D9102E30169822487FDE5DE00A352B01D
-  $ca = $sig | Select-Object -ExpandProperty 'TimeStamperCertificate'
-  $ca_thumbprint = $ca | Select-Object -ExpandProperty 'Thumbprint'
-  if ( $ca_thumbprint -ne '614D271D9102E30169822487FDE5DE00A352B01D' ) {
-    Write-Output "CA Thumbprint failed verification. Has the CA certificate been changed? If so, then please submit a PR to fix this."
-    Write-Output $( $ca | Format-List )
-    Exit 1
-  }
+  if (-Not $s) {
+    $gpg_keyring = "$env:temp\keyring.gpg"
 
-  # Amazon.com certificate is:
-  #  - Subject: 'CN=Amazon.com Services LLC, OU=Software Services, O=Amazon.com Services LLC, L=Seattle, S=Washington, C=US'
-  #  We do not check the Thumbprint because it will change when the certificate is rotated. This seems to be yearly.
-  $cert = $sig | Select-Object -ExpandProperty 'SignerCertificate'
-  $cert_subject = $cert | Select-Object -ExpandProperty 'Subject'
-  if ( $cert_subject -ne 'CN=Amazon.com Services LLC, OU=Software Services, O=Amazon.com Services LLC, L=Seattle, S=Washington, C=US' ) {
-    Write-Output "Amazon.com is not the issuer of this installer. If this is the correct then please submit a PR to fix this."
-    Write-Output $( $cert | Format-List )
-    Exit 1
+    # Download GPG
+    $gpg_installer = "$env:temp\gnupg-w32-2.2.27_20210111.exe"
+    wget https://gnupg.org/ftp/gcrypt/binary/gnupg-w32-2.2.27_20210111.exe -OutFile $gpg_installer
+
+    # Verify GPG
+    $gpg_sig = Get-AuthenticodeSignature $gpg_installer
+    $status = $gpg_sig | Select-Object -ExpandProperty 'Status'
+    if ( $status -ne 'Valid' ) {
+      Write-Output "GPG installer does not have a valid signature."
+      Exit 1
+    }
+    $sha256 = Get-FileHash $gpg_installer -Algorithm SHA256
+    if ( $sha256 -inotmatch '5D89E239790822711EAE2899467A764879D21440AB68E9413452FA96CEDEBA50') {
```

Comment:
nit: Should we store this hash in a variable like we do for `$fingerprint`?

Reply:
Fixed